dask/array/routines.py

```diff
@@ -1146,6 +1146,28 @@ def _int_piecewise(x, *condlist, **kwargs):
     )
 
 
+def _unravel_index_kernel(indices, func_kwargs):
+    return np.stack(np.unravel_index(indices, **func_kwargs))
+
+
+@wraps(np.unravel_index)
+def unravel_index(indices, dims, order='C'):
+    if dims and indices.size:
+        unraveled_indices = tuple(indices.map_blocks(
+            _unravel_index_kernel,
+            dtype=np.intp,
+            chunks=(((len(dims),),) + indices.chunks),
```

Comment:
@mrocklin basically this is the only line that I don't understand. Chunking is still pretty weird for me.

Reply:
Right. So what is happening is we are telling `map_blocks` that we are adding a new axis at position `0` and that axis has 1 chunk, which has size `len(dims)`. The rest of the chunks are unchanged in size. As an example, if we have `indices = [14]` and `dims = (4, 5)`, calling NumPy's `unravel_index` would give us `[(2, 4)]`. As we would get `(2, 4)` from operating on `14`, it is in the same chunk. HTH. Please feel free to ask more questions.

Reply:
I guess I had never actually unraveled more than 1 index. 🦊 

Ok, got it. You even handled the fact that numpy needs this to be a tuple for multiindexing to work correctly. Looks good to me.

Can you add a test where you actually index an array with the output of this function? Part of your PR included using `tuple` to make sure multiindexing would work.

Reply:
Yeah, it can actually be pretty useful if you have N-D array of indices for example.

Sure that sounds reasonable. Will give it a go.

Reply:
Have pushed a test for this. Works for me locally. Will see what CIs say.